File: 5/a.ts
// const input = await Bun.file("test.txt").text();
const input = await Bun.file("input.txt").text();

let res = Infinity;

const seeds = input
  .split("\n")[0]
  .split(": ")[1]
  .split(" ")
  .map((d) => parseInt(d));
// const seeds = [79];

let converters = input
  .split("\n\n")
  .slice(1)
  .map((d) =>
    d
      .split("\n")
      .slice(1)
      .map((d) => d.split(" ").map((d) => parseInt(d)))
  );
// converters = [converters.shift()!];

console.log(seeds);

seeds.forEach((seed) => {
  for (const map of converters) {
    for (const [to, from, l] of map) {
      if (from <= seed && from + l > seed) {
        seed = to + (seed - from);
        break;
      }
    }
  }
  console.log(seed);
  if (seed < res) res = seed;
});

console.log();
console.log(res);

File: 5/a.go
package main

import (
	"fmt"
	"math"
	"os"
	"slices"
	"strconv"
	"strings"
)

type Translation struct {
	to     int
	from   int
	length int
}

func main() {
	file, _ := os.ReadFile("input.txt")

	res := math.MaxInt32

	seedsPart, mapPart, _ := strings.Cut(string(file), "\n\n")
	seeds := slices.Delete(strings.Split(seedsPart, " "), 0, 1)
	steps := strings.Split(mapPart, "\n\n")
	translationSteps := make([][]Translation, len(steps))

	for i, step := range steps {
		lines := slices.Delete(strings.Split(step, "\n"), 0, 1)
		translationSteps[i] = make([]Translation, 0, len(lines))
		for _, line := range lines {
			numbers := strings.Split(line, " ")
			to, _ := strconv.Atoi(numbers[0])
			from, _ := strconv.Atoi(numbers[1])
			length, _ := strconv.Atoi(numbers[2])
			translationSteps[i] = append(translationSteps[i], Translation{to, from, length})
		}
	}

	for _, seed := range seeds {
		num, _ := strconv.Atoi(seed)
		for _, step := range translationSteps {
			for _, translation := range step {
				if num >= translation.from && num <= translation.from+translation.length {
					num = translation.to + (num - translation.from)
					break
				}
			}
		}
		// fmt.Println(num, seed)
		if num < res {
			res = num
		}
	}

	fmt.Println(res)
}
